Proposed new branch Humble - Fix library to work with ROS2 Humble

enrico391 opened the pull request:
Hi, I fix the library to use it with ROS2 Humble, I could be useful for someone like me that are using this version of ROS

enrico391 committed:
Fix library to work with ROS2 Humble

ottojo commented:
Hi! No issues having a humble branch. I will have to check myself, but from the changes you made, the only one i could imagine being version specific seems to be `realtime_publisher.hpp`? And this looks like a change that could be useful for all distros.

Please do check if it works on humble with only that change, and then i will check if that works on jazzy/kilted as well, maybe we can have a jazzy branch which also works for humble and save us some potential backporting issues with maintaining many branches!

enrico391 commented:
Hi, the real problem was related to the context of function on_init(). Maybe I'm missing something but if I launch the broadcaster without checking if the node has the parameters (sensor_name, power_supply_tec, ecc) it crash (I add in the parameter yaml file the correct parameters and the other thingsl like in the documentation).
So I added the parameter checker for that parameters and it works but I don't know exactly why. 
Instead for the hpp header I changed it because there was a deprecation warning.

I agree with you for the potential issues with many branches! If you have some advices or if you could know what is the reason of this problem I'm open to learn! Thanks

enrico391 commented:
I forgot to say that before used in Humble I tried in jazzy with the same setup and it worked without any problems

enrico391 committed:
add percentage estimation

enrico391 committed:
add current publisher
